File: platform-demos/C/image-viewer/image-viewer.py
#!/usr/bin/python

from gi.repository import Gtk, GdkPixbuf, Gdk
import os, sys

class GUI:
	image = 0

	def __init__(self):

		window = Gtk.Window()
		window.set_title ("Image Viewer")
		window.connect_after('destroy', self.destroy)

		box = Gtk.Box()
		box.set_spacing(5)
		box.set_orientation (Gtk.Orientation.VERTICAL)
		window.add (box)

		self.image = Gtk.Image()
		box.pack_start (self.image, True, True, 0)

		button = Gtk.Button ("Open a picture...")
		box.pack_start (button, False, False, 0)
		button.connect_after('clicked', self.on_open_clicked)

		window.show_all()

	def destroy(window, self):
		Gtk.main_quit()

	def on_open_clicked (self, button):
		dialog = Gtk.FileChooserDialog ("Open Image", button.get_toplevel(), Gtk.FileChooserAction.OPEN);
		dialog.add_button (Gtk.STOCK_CANCEL, 0)
		dialog.add_button (Gtk.STOCK_OPEN, 1)
		dialog.set_default_response(1)

		filefilter = Gtk.FileFilter ()
		filefilter.add_pixbuf_formats ()
		dialog.set_filter(filefilter)

		if dialog.run() == 1:
			self.image.set_from_file(dialog.get_filename())
		dialog.destroy()

def main():
	app = GUI()
	Gtk.main()

if __name__ == "__main__":
    sys.exit(main())

File: platform-demos/C/image-viewer/image-viewer.js
const Gtk = imports.gi.Gtk;
const Lang = imports.lang;

function ImageViewer () {
  this._init ();
}

ImageViewer.prototype = {
  _init: function () {
    this.window = new Gtk.Window ({title: "Image Viewer Demo"});
    this.window.connect ("hide", Gtk.main_quit);

    var main_box = new Gtk.Box ({orientation: Gtk.Orientation.VERTICAL, spacing: 0});
    this.window.add (main_box);

    this.image = new Gtk.Image ();
    main_box.pack_start (this.image, true, true, 0);

    var open_button = new Gtk.Button ({label: "Open a picture..."});
    open_button.connect ("clicked", Lang.bind (this, this._openClicked));
    main_box.pack_start (open_button, false, false, 0);

    this.window.show_all ();
  },

  _openClicked: function () {
    var chooser = new Gtk.FileChooserDialog ({title: "Select an image",
                                              action: Gtk.FileChooserAction.OPEN,
                                              transient_for: this.window,
                                              modal: true});
    chooser.add_button (Gtk.STOCK_CANCEL, 0);
    chooser.add_button (Gtk.STOCK_OPEN, 1);
    chooser.set_default_response (1);
    
    var filter = new Gtk.FileFilter ();
    filter.add_pixbuf_formats ();
    chooser.filter = filter;

    if (chooser.run () == 1)
      this.image.file = chooser.get_filename ();

    chooser.destroy ();
  }
}

Gtk.init (null, null);

var iv = new ImageViewer ();

Gtk.main ();
